app/src/main/java/org/indywidualni/fblite/activity/MainActivity.java

```diff
@@ -430,6 +444,28 @@ private boolean requestStoragePermission() {
     }
 
     @Override
+    public void onCreateContextMenu(ContextMenu menu, View view, ContextMenu.ContextMenuInfo menuInfo) {
+        WebView.HitTestResult result = webView.getHitTestResult();
+        if (result != null) {
+            int type = result.getType();
+
+            if (type == WebView.HitTestResult.IMAGE_TYPE || type == WebView.HitTestResult.SRC_IMAGE_ANCHOR_TYPE) {
+                showLongPressedImageMenu(menu, result.getExtra());
+            }
+        }
+    }
+
+    @Override
+    public boolean onContextItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
```

Comment:
One can later add other menu items, like "Share url" and "Share image", with latter perhaps being possible directly from WebView cache.